public/app/plugins/datasource/grafana-azure-monitor-datasource/components/QueryEditor/useDefaultQuery.ts

```diff
@@ -0,0 +1,35 @@
+import { useEffect, useMemo } from 'react';
+import { AzureMonitorQuery, AzureQueryType } from '../../types';
+
+const DEFAULT_QUERY_TYPE = AzureQueryType.AzureMonitor;
+
+const createQueryWithDefaults = (query: AzureMonitorQuery) => {
+  // A quick and easy way to set just the default query type. If we want to set any other defaults,
+  // we might want to look into something more robust
+  if (!query.queryType) {
+    return {
+      ...query,
+      queryType: query.queryType ?? DEFAULT_QUERY_TYPE,
+    };
+  }
+
+  return query;
+};
+
+/**
+ * Returns queries with some defaults, and calls onChange function to notify if it changes
+ */
+const useDefaultQuery = (query: AzureMonitorQuery, onChangeQuery: (newQuery: AzureMonitorQuery) => void) => {
+  const queryWithDefaults = useMemo(() => createQueryWithDefaults(query), [query]);
+
+  useEffect(() => {
+    if (queryWithDefaults !== query) {
+      console.log('changing default query');
```

Comment:
why doesn't our linter yell at us about console.logs?

Reply:
i was just wondering that myself.......  I'm sick of accidentally committing these in 😭